p2p/proto/protocols.md

```diff
@@ -1,11 +1,259 @@
+types, their content and their hash calculation see [here](https://docs.starknet.io/documentation/architecture_and_concepts/Network_Architecture/transactions/).
+
+The commitment of the transactions in the header is comprised of the number of transactions and
+their merkle root.
+
+In order to verify the transactions, the node needs to calculate their hash and compare it to the
+transactions merkle root in the block header.
+
+The number can be used to delimiter between the transactions of each block. 
+
+In order for the outbound peer to know that the inbound peer is malicious as soon as possible, they
+should do two things:
+
+1. Calculate the transaction hash once each transaction arrives and validate it's the same hash.
+Even though we can only verify that the hash appears in the commitment after we got all the
+transactions for the block, this check is a proof of work. It checks that the inbound peer did some
+work at some point of time to calculate the hash of this transaction and delays a peer that sends
+transactions with random fields
+2. Reject the connection and declare the peer malicious once it sent more transactions than the sum
+of transaction amounts in the headers of the requested blocks.
+
+### Events
+The Events protocol is used to download the events emitted in a range of blocks.
+
+Its name for negotiation is `/starknet/events/1`
+
+Each single message is a fin or an [Event](./event.proto)
+
+Each event contains
+* The index of the transaction that emitted this event
+(If it's the first transaction in the block, the index will be 0).
+* The address of the contract that emitted this event.
+* The keys and data of this event.
+
+The commitment of the events in the header is comprised of the number of events and
+their merkle root.
+
+In order to verify the events, the node needs to calculate their hash and compare it to the
+events merkle root in the block header.
+
+The number can be used to delimiter between the events of each block, and the transaction index field
+of each event is used to delimiter between the events of each transaction within a block.
+
+**Important Note**: The transaction index is currently not part of the headers hash. Until this is
+fixed, the value of the `transaction_index` field in each event is trusted.
+
+In order for the outbound peer to know that the inbound peer is malicious as soon as possible, they
+should do the same two things recommended in the [transactions](#transactions) protocol.
+
+### Receipts
+The Receipts protocol is used to download the receipts of transactions in a range of blocks.
+
+Its name for negotiation is `/starknet/events/1`
+
+Each single message is a fin or a [Receipt](./receipt.proto)
+
+There is a single receipt per transaction.
+
+This means that the number of transactions inside the header can be used to delimiter between the
+receipts of each block.
+
+#### Important Note
+The receipts commitment in the header is currently not part of the block hash.
```

Comment:
It's not part of the block hash the feeder returns. The block header does contain them and I think it's hash should be the P2P block hash. The nodes that initially publish it will trust the feeder for the receipts (as they trust it on the block hash anyway)